lib/common/commonUtil.py

```diff
@@ -560,6 +562,104 @@ def deep_merge_dict(source_dict, destination_dict):
         return destination_dict
 
 
+class NetworkUtil(object):
+
+    @staticmethod
+    def download_file(output_dp, download_link):
+        logger.info("Prepare to download the file from link [%s]" % download_link)
+        response = requests.get(download_link, verify=False, stream=True)
+        download_fn = download_link.split("/")[-1]
+        if os.path.exists(output_dp) is False:
+            os.makedirs(output_dp)
+        download_fp = os.path.join(output_dp, download_fn)
+        try:
+            try:
+                total_len = int(response.headers['content-length'])
+            except:
+                total_len = None
+            with open(download_fp, 'wb') as fh:
+                for data in tqdm(response.iter_content(chunk_size=512 * 1024), total=total_len / (512 * 1024)):
+                    fh.write(data)
+            return download_fp
+        except Exception as e:
+            logger.error("ERROR: [%s]" % e.message)
+            return None
+
+    @staticmethod
+    def get_files_from_remote_url_folder(input_url, input_params=None):
+        return_dict = {}
+        try:
+            if input_params:
+                response_obj = requests.get(input_url, params=input_params)
+            else:
+                response_obj = requests.get(input_url)
+            if response_obj.status_code == 200:
+                for line in response_obj.iter_lines(decode_unicode=True):
+                    match = re.search(r'(?<=href=").*?(?=")', line)
+                    if match:
+                        href_link = match.group(0)
+                        f_name = href_link.split("/")[-1]
+                        return_dict[f_name] = href_link
+            else:
+                logger.error("fetch remote file list error with code [%s]" % str(response_obj.getcode()))
+        except Exception as e:
+            logger.error("ERROR: [%s]" % e.message)
+        return return_dict
+
+    @staticmethod
+    def post_request_and_response(input_url, input_data=None, input_headers=None):
+        """
+
+        @param input_url: url
+        @param input_data: post data
+        @param input_headers: post header
+        @return: response obj, you can get json obj by using response_obj.json()
+        """
+        try:
+            if input_data and input_headers:
+                response_obj = requests.post(input_url, data=input_data, headers=input_headers)
+            elif input_data:
+                response_obj = requests.post(input_url, data=input_data)
+            elif input_headers:
+                response_obj = requests.post(input_url, headers=input_headers)
+            else:
+                response_obj = requests.post(input_url)
```

Comment:
due to the default value of `input_data` and `input_headers` are `None`, it seems like we can just call `response_obj = requests.post(input_url, data=input_data, headers=input_headers)`.

Reply:
without giving parameters behave differently with specifying the parameter with None value

Reply:
wow, it's really mysterious.
base on the code [api.py](https://github.com/requests/requests/blob/v2.18.4/requests/api.py#L16-L58) and [sessions.py](https://github.com/requests/requests/blob/v2.18.4/requests/sessions.py#L441-L444), the default value should be None 🤔